spec/classes/redis_spec.rb

```diff
@@ -1329,6 +1338,34 @@ class { 'redis::globals':
         end
 
         it { is_expected.to contain_file(service_file) }
+
+        it do
+          content = <<-END.gsub(%r{^\s+\|}, '')
+            |[Unit]
+            |Description=Redis Advanced key-value store for instance default
+            |After=network.target
+            |After=network-online.target
+            |Wants=network-online.target
+            |
+            |[Service]
+            |RuntimeDirectory=redis
+            |RuntimeDirectoryMode=2755
+            |Type=#{servicetype}
+            |ExecStart=/usr/bin/redis-server #{config_file}#{systemd}
+            |ExecStop=/usr/bin/redis-server -p 6379 shutdown
+            |Restart=always
+            |User=redis
+            |Group=redis
+            |LimitNOFILE=65536
+            |
+            |[Install]
+            |WantedBy=multi-user.target
+          END
+
+          if facts['service_provider'] == 'systemd'
```

Comment:
I always like to add an `else is_expected.not_to contain_file(service_file)` to these kind of situations.

Reply:
Sure, will add it.

Reply:
Actually, `not_to` would be wrong here. This test also works for non-systemd systems, but apparently with different content. I've added a small test for non-systemd systems instead.